react/README.md

```diff
@@ -1259,34 +1267,338 @@ For more info on `dangerouslySetInnerHTML`, see: [Dangerously Set innerHTML](htt
+        })
+    }
+
+    render() {
+        let {visible} = this.state;
+        let messageClassName;
+
+        if (!visible) {
+            messageClassName = 'hidden';
+        }
+
+        return (
+            <div>
+                <Button click={this._handleToggle.bind(this)}>Toggle!</Button>
+                <p className={messageClassName}>
+                    This message is toggled on/off with CSS 🙁!
+                </p>
+            </div>
+        );
+    }
+}
+```
+
+_Friendly reminder:_ If you want an **entire** component to be conditionally rendered, the component must return `null`. Returning `undefined` will be an error.
 
 **[⬆ back to top](#table-of-contents)**
 
 ## State
 
+There are two ways of maintaining data in a React component: props and state.
+
+The props are used by a component's parent to configure the component and are immutable within the component.
+The component's state is internal to the component so that it can maintain data that will change over time. Whenever the state changes (via `setState`), the component will re-render. A component's state should not be manipulated outside of it.
+
 ### Initializing
 
-Coming soon...
+We rely on the [Class fields proposal](https://github.com/jeffmo/es-class-fields-and-static-properties) for initialize state over assigning `this.state` in the constructor so that it's more declarative. We don't use `getInitialState` which was the only option in ES5.
 
-**[⬆ back to top](#table-of-contents)**
+```js
+// good
+export default class Togglr extends React.Component {
+    state = {visible: false}
 
-### Dynamic data
+    // rest of the component
+}
+
+// bad (assigns `this.state` unnecessarily in constructor)
+export default class Togglr extends React.Component {
+    constructor(props, context) {
+        super(props, context);
+        this.state = {visible: false};
+    }
+
+    // rest of the component
+}
 
-Coming soon...
+// bad (uses ES5 `getInitialState`)
+export default class Togglr extends React.Component {
+    getInitialState() {
+        return {visible: false};
+    }
+
+    // rest of the component
+}
+```
 
 **[⬆ back to top](#table-of-contents)**
 
 ### Defaulting from props
 
-Coming soon...
+In general, using props to generate state is an anti-pattern because it results in duplicate "sources of truth" (see [Props in getInitialState as an anti-pattern](https://facebook.github.io/react/tips/props-in-getInitialState-as-anti-pattern.html)). But if your props is properly named to indicate that it's only used as seed data for the component's internally-controlled state, it's no longer an anti-pattern.
+
+We tend to prefix these types of props with `default*` to match the `defaultValue` prop React uses for input elements. `initial*` is also a good prefix.
+
+When defaulting from props, using the declarative class field no longer works as the props are not available in static scope. The only option is to initialize within the `constructor`:
```

Comment:
I know that it is mentioned but maybe emphasize further that this style of initialization is only okay when utilizing default props. Previous example saying this is something that shouldn't be done may cause confusion.